### DISCUSSION TITLE
Improving speed of writing

### DISCUSSION THREAD

**Discussion by tuongbuimtt** [yosh-matsuda/cpp-yyjson#17] [Q&A]
For exampl to my situation:
```
object json_nested_1;
// process adding some field
object json_nested_2;
// process adding some field
object json;
// checking  some conditions: json_nest_1 and json_nest_2
// only emplacing a json_nest_1
json.emplace("abc", json_nested_1);
```
I take some practices with yyjson to replacing current rapidjson. But my results of yyjson is slower than rapidjon.
Do you have any advice to me to improve speed of creating json? @yosh-matsuda 
Noted: 
I also try to call
```
auto nested_json = json.emplace("xyz", empty_object);
nested_json.emplace("123", 123);
```
as much as possible.

**Comment by yosh-matsuda** [comment]
Can you do "checking some conditions" before creating `json_nested_1` and `json_nested_2`? Creating a new JSON object may be costly. Do you have a reproducible example?

If the object consists of many strings, the difference between rapidjson and yyjson seems to be at most twice as large. It would be small depending on the situation.  
https://user-images.githubusercontent.com/59041398/222316476-293d8081-d7f5-4f70-b0ea-013ac4bc7943.png  
Benchmark code: https://github.com/yosh-matsuda/cpp-yyjson/blob/main/test/bench_write.cpp#L932

**Reply by tuongbuimtt** [reply]
Update: I took a long time to rewrite my code to apply cppyyjson, limiting to create new object, the lastest result is equal to rapidjson.

**Comment by tuongbuimtt** [comment]
Is there any support creating json object from specific allocator like Rapidjson?
```
// Rapid json
Docment doc;
Value nest_val = Value("ABC", doc.GetAllocator());
doc.Addmember("abc", nest_val);
```
I expect ability to create new JSON object from specific the father Object allocator. Maybe this faster?

**Reply by yosh-matsuda** [reply]
I added that the allocator can be accepted in `write` functions in v0.5.0. The new `dynamic_allocator` is suitable for `write` because the size of the JSON string has not been known before.

However, the speedup may only be for large strings. My benchmarks show about a 10% increase for large JSON strings (e.g., over 10MB) but no change for smaller sizes. This would be related to the CPU cache size.
